Assets/Character/AaronTestAssets/PlayerMovement.cs

```diff
@@ -6,79 +6,108 @@
 public class PlayerMovement : MonoBehaviour {
 
 
-    public float speed = 2f;
-    public float jump = 8f;
+    public float move_speed = 2f;
+    public float jump_power = 8f;
+    public float roll_speed = 4f;
+    public float roll_time = 0.5f;
+
+
     private Animator animator;
     private Rigidbody2D rigidbody;
-    private bool isFacingRight = true;
-	// Use this for initialization
+    private int directionFacing = 1;
+    //private ArrayList lockStates = new ArrayList();
+    private AnimatorStateInfo stateInfo;
+    private Vector3 scale;
+    private float timer;
+	
+
 	void Start () {
 
         animator = GetComponent<Animator>();
         rigidbody = GetComponent<Rigidbody2D>();
-
-        // if (animator.)
+       // lockStates.Add();
+        stateInfo = animator.GetCurrentAnimatorStateInfo(0);
+        scale = transform.localScale;
 
     }
 	
-	// Update is called once per frame
 	void FixedUpdate () {
-
-        float horizontal = (Input.GetAxis("Horizontal"));
-
-        //position based
-        transform.position += new Vector3(horizontal, 0, 0) * speed * Time.deltaTime;
-
-        //add force
-        //rigidbody.AddForce(new Vector2(horizontal * speed, 0), ForceMode2D.Force);
-
-        if(animator.GetBool("OnGround"))
+        if (!animator.GetBool("Rolling"))
```

Comment:
States are exclusive, so there are better ways to do this. You can write a C# class for each state that handles the input or event to transition to another state. and have the player attached script map its update to it's current state's update. You can also see how Christian implemented his on his branch or look up other ways to do it.